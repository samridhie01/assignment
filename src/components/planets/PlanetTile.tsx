import React from "react";
import { Link } from "react-router-dom";
import { makeStyles, Typography, Box } from "@material-ui/core";
import {ReactComponent as PlanetIcon} from "../../assets/planet.svg";

const useStyles = makeStyles({
  flexContainer2: {
    display: "flex",
    flexFlow: "column wrap",
    alignContent: "center"
  },
  item: {
    display: "flex",
    flex: "0 0 0",
    width: "100%",
    alignContent: "center"
  },
  
});

interface PlanetData {
  name: string;
  climate: string;

}

const PlanetTile: React.FC<PlanetData> = ({ name, climate }) => {
  const classes = useStyles();
  return (
    <>
       <PlanetIcon />
      <div className={classes.flexContainer2}>
        <h3>{name}</h3>
        <Box className={classes.item}>
        <Typography variant="subtitle2" >{climate} climate</Typography>
        </Box>
        <br/>
        <Link to={{ pathname: "/details", search: `?search=${name}` }}>
          View details
        </Link>
      </div>
    </>
  );
};

export default PlanetTile;
