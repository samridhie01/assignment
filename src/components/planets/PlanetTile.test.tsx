import {render} from "@testing-library/react";

import React from "react";
import PlanetTile from "./PlanetTile";
import {BrowserRouter} from "react-router-dom";

describe("Search Box", ()=>{
    it('should render snapshot', () => {
        const { asFragment } = render(
            <BrowserRouter>
            <PlanetTile name={"abc"} climate={"def"} />
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
})
