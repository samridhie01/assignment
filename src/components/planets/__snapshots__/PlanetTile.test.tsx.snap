// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box should render snapshot 1`] = `
<DocumentFragment>
  <svg>
    planet.svg
  </svg>
  <div
    class="makeStyles-flexContainer2-1"
  >
    <h3>
      abc
    </h3>
    <div
      class="MuiBox-root MuiBox-root-3 makeStyles-item-2"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle2"
      >
        def climate
      </h6>
    </div>
    <br />
    <a
      href="/details?search=abc"
    >
      View details
    </a>
  </div>
</DocumentFragment>
`;
